relayer/provider/provider.go

```diff
@@ -47,8 +49,8 @@ func (r RelayerTxResponse) MarshalLogObject(enc zapcore.ObjectEncoder) error {
 type KeyProvider interface {
 	CreateKeystore(path string) error
 	KeystoreCreated(path string) bool
-	AddKey(name string) (output *KeyOutput, err error)
-	RestoreKey(name, mnemonic string) (address string, err error)
+	AddKey(name string, coinType uint32) (output *lens.KeyOutput, err error)
```

Comment:
Instead of relayed declaring an interface directly referencing another module, I had an adapter approach in https://github.com/cosmos/relayer/commit/18f8146d65f8a5b24adcfeb141e57da77fee70d1. WDYT?

Reply:
Your approach seems preferable! I can get that pulled into this PR